core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -6222,4 +6222,62 @@ class KafkaApisTest {
 
     assertEquals(expectedResponse, response.data)
   }
+
+  @Test
+  def testGetTelemetrySubscriptionsNotAllowedForZkClusters(): Unit = {
+    val data = new GetTelemetrySubscriptionsRequestData()
+    data.setClientInstanceId(Uuid.randomUuid())
+
+    val request = buildRequest(new GetTelemetrySubscriptionsRequest.Builder(data, true).build())
+    createKafkaApis(enableForwarding = true).handle(request, RequestLocal.NoCaching)
+
+    val response = verifyNoThrottling[GetTelemetrySubscriptionsResponse](request)
+    assertEquals(Errors.UNKNOWN_SERVER_ERROR, Errors.forCode(response.data.errorCode))
```

Comment:
Test name says `NotAllowedForZkCluster`, so why do we get `UNKNOWN_SERVER_ERROR` back?

Reply:
True but KafkaApis.scala returns IllegalStateException when respective API is not enabled. The reason it returns UNKNOWN_SERVER_ERROR as this path should not be encountered in practice as prior check from ApiVersion response should list which all APIs are enabled.

https://github.com/apache/kafka/blob/e7e399b9409b42f82d7ce57b99a461c465e5849d/core/src/main/scala/kafka/server/KafkaApis.scala#L176

Reply:
If this case cannot happen in practice, what do we actually test?

Reply:
I mean the path should not occur means that the clients should not issue RPC for ZK broker when not enabled.
This test verifies following behaviour: That we have correctly not enabled ZK broker in RPC json hence if client mistakenly issues RPC over ZK broker then they should fail.

Though I agree there could have been a better error mapped in broker for this but seems out of scope for this PR. May be someone from server side can provide more context.